### PR TITLE
Parametric curves support

### LINKED ISSUE #91: Construction of edges based on parametric equations
Handy for creation of mathematically defined shapes like airfoils, cycloidal gears , spirals

This would be a general solution for  #37

### PR DESCRIPTION
**Pull request by adam-urbanczyk**
This will resolve #91 . Also in principle #37 could be solved by this PR.

 Sneak preview of the functionality:
```python
import cadquery as cq
from math import sin, cos,pi,floor

def hypocycloid(t,r1,r2):
    return ((r1-r2)*cos(t)+r2*cos(r1/r2*t-t),(r1-r2)*sin(t)+r2*sin(-(r1/r2*t-t)))

def epicycloid(t,r1,r2):
    return ((r1+r2)*cos(t)-r2*cos(r1/r2*t+t),(r1+r2)*sin(t)-r2*sin(r1/r2*t+t))

def gear(t,r1=2,r2=1):
    if (-1)**(1+floor(t/2/pi*(r1/r2))) < 0:
        return epicycloid(t,r1,r2)
    else:
        return hypocycloid(t,r1,r2)

result = cq.Workplane('XY').parametricCurve(lambda t: gear(t*2*pi,6,1)).twistExtrude(15,90)
show_object(result)
```
afbeelding

### CODE REVIEW

**Comment by adam-urbanczyk** [comment]
I think it is ready for review. I made quite some changes to the spline function so please take a good look.

I also added one example to the docs.

File: cadquery/cq.py [hunk 1]
```diff
@@ -1312,14 +1312,16 @@ def move(self, xDist=0, yDist=0):
         return self.newObject([self.plane.toWorldCoords(newCenter)])
 
     def spline(self, listOfXYTuple, tangents=None, periodic=False,
-               forConstruction=False):
+               forConstruction=False, includeCurrent=True, makeWire=False):
```
Comment (jmwright) [inline cadquery/cq.py]: I'm thinking the default values of the two new parameters have the correct defaults to make sure spline works as it always has. If so, I think this change is fine.

Comment (adam-urbanczyk) [reply cadquery/cq.py]: Indeed. Spline tests pass without any modifications.

**Comment by dcowden** [comment]
+1 for this, looks great and adds hotness too.

**Comment by adam-urbanczyk** [comment]
@dcowden +1 for hotness : ). Merging.

**Pull request merged**